src/borg/crypto/key.py

```diff
@@ -170,6 +170,10 @@ class KeyBase:
     # was supplied, and if an empty passphrase works, then Borg won't ask for one.
     logically_encrypted = False
 
+    # skip the check for valid IDs.
+    # This is required after loading data from a repo with an alternative key.
```

Comment:
... after "borg load" because data is stored to IDs which are valid at origin repo, but not here.